pytups/tuplist.py

```diff
@@ -100,14 +100,33 @@ def to_dict(
         """
         from . import superdict as sd
 
-        if result_col is None:
+        if len(self) == 0:
+            return sd.SuperDict()
+        # Handle case of list of dict
+        if isinstance(self[0], dict):
+            if indices is None:
+                raise ValueError(
```

Comment:
in fact, this check can be done inside the line that already exists with `if indices is None:`

Reply:
I do not think that there is a perfect way to do things without so many if...else or to order them more logically.
The order of the checks matter and can be different for dict or tuple.

for dict, I need to check for indices =None first because the case of result_col = None uses indices.
for tuples, I need to check for result_col=None first because the case of indices=None uses result_col.